src/aposteriori/data_prep/create_frame_data_set.py

```diff
@@ -1165,11 +1174,54 @@ def _fetch_pdb(
     if len(pdb_code) == 5:
         # Extract chain from string:
         chain = pdb_code[-1]
-        _ = _select_pdb_chain(output_path, chain)
+        _, output_path = _select_pdb_chain(output_path, chain)
 
     return output_path
 
 
+def filter_structures_by_blacklist(
+    structure_files: t.List[pathlib.Path], blacklist_csv_file: pathlib.Path
+) -> t.List[StrOrPath]:
+    """
+    Filters structures contained in a blacklist csv file.
+
+    Parameters
+    ----------
+    structure_files: List[pathlib.Path]
+        List of paths to pdb files to be processed into frames
+    blacklist_csv_file: pathlib.Path
+        Path to blacklist csv.
+
+    Returns
+    -------
+    filtered_structure_files: List[pathlib.Path]
+        List of filtered paths to pdb files to be processed into frames
+    """
+
+    # Open blacklist file and extract list:
+    with open(blacklist_csv_file) as csv_file:
+        blacklist_csv = csv.reader(csv_file, delimiter=",")
+        # Reading to set to make sure entries are unique:
+        blacklist = set([b_pdb.strip(" ").lower() for b_pdb in next(blacklist_csv)])
```

Comment:
```suggestion
        blacklist = {b_pdb.strip(" ").lower() for b_pdb in next(blacklist_csv)}
```
A set comprehension will save memory here as you won't write to an intermediate list.